src/citrine/_rest/collection.py

```
@@ -222,3 +228,60 @@ def _check_experimental(self, data):
+            The job submission object or job ID that was given from a job submission.
+        timeout
+            Amount of time to wait on the job (in seconds) before giving up. Defaults
+            to 2 minutes. Note that this number has no effect on the underlying job
+            itself, which can also time out server-side.
+
+        Returns
+        -------
+        JobStatusResponse
+            The job response information that can be used to extract relevant
+            information from the completed job.
+
+        """
+        if isinstance(job, JobSubmissionResponse):
+            job_id = job.job_id
+        else:
+            job_id = job  # pragma: no cover
+        path = 'projects/{}/execution/job-status'.format(project_id)
+        params = {'job_id': job_id}
+        start_time = time()
+        while True:
+            response = self.session.get_resource(path=path, params=params)
+            status: JobStatusResponse = JobStatusResponse.build(response)
+            if status.status in ['Success', 'Failure']:
+                break
+            elif time() - start_time < timeout:
+                logger.info('Job still in progress, polling status again in {:.2f} seconds.'
+                            .format(polling_delay))
+
+                sleep(polling_delay)
+            else:
+                logger.error('Job exceeded user timeout of {} seconds.'.format(timeout))
+                logger.debug('Last status: {}'.format(status.dump()))
+                raise TimeoutError('Job {} timed out.'.format(job_id))
+        if status.status == 'Failure':
+            logger.debug('Job terminated with Failure status: {}'.format(status.dump()))
+            for task in status.tasks:
+                if task.status == 'Failure':
+                    logger.error('Task {} failed with reason "{}"'.format(
+                        task.id, task.failure_reason))
+            raise RuntimeError('Job {} terminated with Failure status.'.format(job_id))
```

Could we raise a custom exception here (derived from `RuntimeError`), e.g. [these](https://github.com/CitrineInformatics/citrine-python/blob/master/src/citrine/exceptions.py).

---

Should this message include the `task.failure_reason`?

---

A custom exception can work, and I think including the task failure reason is appropriate.

---

I've also included that info (and the job ID).